src/sagas/pushNotification.js

```diff
@@ -444,6 +445,28 @@ export function* dismissOptInQuestion() {
   yield STORE.setItem(pushNotificationKey.optInDismissed, true);
 }
 
+const cleanToken = async () => {
+  await messaging().unregisterDeviceForRemoteMessages();
+  await messaging().deleteToken();
+};
+
+/**
+ * This function is responsible for reset the push notification.
+ */
+export function* resetPushNotification() {
+  // Unregister the device from FCM
+  yield call(cleanToken);
+  // Clean the store
+  yield STORE.removeItem(pushNotificationKey.enabledAt);
+  yield STORE.removeItem(pushNotificationKey.settings);
+  yield STORE.removeItem(pushNotificationKey.hasBeenEnabled);
+  yield STORE.removeItem(pushNotificationKey.optInDismissed);
+  yield STORE.removeItem(pushNotificationKey.deviceId);
+  // Reset the state
+  yield put(pushReset());
+  yield fork(onAppInitialization);
+}
```

Comment:
Don't you need to also remove the push notifications settings from the wallet service?

Reply:
No, it is not necessary because we have a process to clean stale registers. And if the user re-enables the push notification this register will be updated.

And I did this way to avoid bothering the user with another pin prompt in the reset process.